### PR TITLE
Improve Docker developer experience

### PR DESCRIPTION
**Pull request by Swingline0**
During work on #83, I made a few changes that I wanted to share in case they would be useful. Namely:

- 721ca21 adds a `.dockerignore` file to increase image build efficiency. The biggest impact this has is that it prevents copying the host system's bulky `node_modules` into the Docker context, drastically reducing the time required to build the image.

- b3928fd adds an NPM script which builds the docker image and runs the integration test command with the snapshots directory mounted into the container. The goal of this change was to shorten the [steps required](https://github.com/NimaSoroush/differencify/blob/master/CONTRIBUTING.md#steps) for running and updating tests. I'm not sure the NPM script name, `test:integration-in-docker`, is ideal but happy to update it if you'd like. 

Thanks!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
     "build-docs": "mkdir -p docs && jsdoc2md src/*.js > docs/api.md",
     "pretest": "npm run lint",
     "test": "jest --testPathIgnorePatterns integration.test",
-    "test:integration": "jest --testPathPattern integration.test"
+    "test:integration": "jest --testPathPattern integration.test",
```
Comment (NimaSoroush) [inline package.json]: `test:integration` suppose to happen only in docker as the image generated in local machine doesn not match the reference one. So I am happy to completely replace this command. So

```
"test:integration": "docker build -t differencify . && docker run --volume \"$(pwd)/src/integration.tests\":/differencify/src/integration.tests -t  differencify npm run test:integration"
```

**Comment by NimaSoroush** [comment]
Hi @Swingline0, I was always thinking to improve this but never had a chance to do it. Thanks for putting this together

**Comment by Swingline0** [comment]
You bet, @NimaSoroush! I suppose with the new NPM script, the contribution guidelines could also be updated. All that said, I would love if you could give this command a test on your end just to make sure it all works as expected.

**Comment by NimaSoroush** [comment]
I just tried that script locally. I think we still need another script to trigger the integration test only

```
...
    "jest:integration": "jest --testPathPattern integration.test",
    "test:integration": "docker build -t differencify . && docker run --volume \"$(pwd)/src/integration.tests\":/differencify/src/integration.tests -t  differencify npm run jest:integration"
...
```

**Comment by NimaSoroush** [comment]
Long term plan would be great to have everything as docker-compose

**Comment by NimaSoroush** [comment]
Ok, I see what you mean by that. Let's stick with docker for now then.

Thanks for updating. All LGTM, will merge

**Comment by NimaSoroush** [comment]
Ahh, We need to update circleCi to not run test:integration and instead run jest:integration

**Comment by NimaSoroush** [comment]
Otherwise CI failing https://circleci.com/gh/NimaSoroush/differencify/179

**Comment by Swingline0** [comment]
Updated circleCI script, and bumped the version 😀

**Comment by Swingline0** [comment]
Uggghhhh... the paths are different in the detailed output! 

image

**Comment by NimaSoroush** [comment]
So I assume a new PR will all fixes, right?

**Comment by Swingline0** [comment]
All good now via bee9df5a86884872712bdd1d2d3eb4b0a5039dba

**Comment by NimaSoroush** [comment]
Yeah. All good now. Thanks a lot @Swingline0 

I will just publish the latest changes to npm

**Comment by Swingline0** [comment]
Thanks!

**Comment by NimaSoroush** [comment]
https://www.npmjs.com/package/differencify

**Pull request merged**